common/src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -509,6 +510,33 @@ class SchemaService(private val options: Neo4jOptions, private val driverCache:
       .asBoolean()
   }
 
+  def validateQuery(query: String, expectedQueryTypes: org.neo4j.driver.summary.QueryType*): String = try {
+    val queryType = session.run(s"EXPLAIN $query").consume().queryType()
+    if (expectedQueryTypes.isEmpty || expectedQueryTypes.contains(queryType)) {
+      ""
+    } else {
+      s"Query non valid because the expected type should be [${expectedQueryTypes.mkString(", ")}], but the actual type is $queryType"
+    }
+  } catch {
+    case e: Throwable => s"Query not compiled for the following exception: ${ExceptionUtils.getMessage(e)}"
```

Comment:
same question as above

Reply:
In this case the query cames from the compiler itself so it's already included. Please check the tests

Reply:
I'm on the fence, since we do not really own the error message and the query could be removed there, but I guess we can keep it like this for now.